### ISSUE TITLE
More powerful list command

### ISSUE THREAD

**Issue by frostming** [pdm-project/pdm#531]
## Describe the solution you'd like

Add the following options:

- [x] `--json` to output the dep graph as JSON document, must be combined with `--graph`.
- [x] #533

**Comment by ElijahSink** [comment]
- Is there a schema for this JSON document?
- Should the output be pretty-printed?

**Comment by frostming** [comment]
@ElijahSink An example should be like:

```json
[
  {
    "name": "flask",
    "version": "2.0.1",
    "required": "~=2.0",
    "dependencies": [
      {
        "name": "click",
        "version": "8.0.1",
        "required": ">=7.1.2"
      },
      {
        "name": "itsdangerous",
        "version": "2.0.1",
        "required": ">=2.0"
      },
      {
        "name": "jinja2",
        "version": "3.0.1",
        "required": ">=2.0",
        "dependencies": [
          {
            "name": "markupsafe",
            "version": "2.0.1",
            "required": ">=2.0"
          }
        ]
      },
      {
        "name": "werkzeug",
        "version": "2.0.1",
        "required": ">=2.0"
      }
    ]
  }
]
```

**Comment by frostming** [comment]
Here are some arguments about the field names (Naming is so hard :(

1. `package` vs `name`
2. `required` & `requires` (reverse) ?
3. Names of dependencies, three pairs(forward, reverse):
    1. `depenencies` & `dependants`
    2. `requires` & `required_by`
    3. `needs` & `needed_by`

**Comment by frostming** [comment]
Closed by #583
